### DISCUSSION TITLE
Debug & Traffic Admin Interface

### DISCUSSION THREAD

**Discussion by tholzheim** [elimintz/justpy#483] [Ideas]
It would be nice to have a admin interface to observe the
* open sockets
* routing
* traffic - e.g. latency/thruput
* sent components 
* etc. 

to debug and monitor your applications. Currently it is a black box that requires in-depth debugging to observe the traffic and arising issues.

**Comment by WolfgangFahl** [comment]
Indeed this would be a great tool. Especially if we treat this as an "eat your own dog food" app and a justpy app is used for monitoring which runs on a defined port and an be optionally started with the main application or even monitor multiple applications on different ports.

**Comment by tholzheim** [comment]
Since the justpy webpage is based on Starlette maybe there already exists such a monitor tool for Starlette that could be adjusted to our needs

**Reply by Phonix88** [reply]
They don't think that's what you meant, but I stumbled upon this:

https://www.reddit.com/r/Python/comments/xbz8jd/starletteadmin_simple_and_extensible_admin/?utm_medium=android_app&utm_source=share

**Reply by WolfgangFahl** [reply]
@Phonix88  thanks for the link that looks very new it's available at:
https://github.com/jowilf/starlette-admin

the idea of that project is interesting but different from what we are up to here. Still its worthwhile for comparison of the approaches. starlette-admin uses templates heavily for the UI while justpy has the approach to specify the ui mostly in code.

**Comment by WolfgangFahl** [comment]
https://stackoverflow.com/questions/552744/how-do-i-profile-memory-usage-in-python might help

**Comment by WolfgangFahl** [comment]
see also https://github.com/mchlvl/starlette-zipkin
